packages/data-point/lib/reducer-types/reducer-object/factory.js

```diff
@@ -26,42 +32,61 @@ function isType (source) {
 
 module.exports.isType = isType
 
+/**
+ * @return {Object}
+ */
+function newProps () {
+  return {
+    constants: {},
+    reducers: []
+  }
+}
+
+module.exports.newProps = newProps
+
 /**
  * @param {Function} createReducer
  * @param {Object} source
- * @param {Array} path
- * @param {Array} props
+ * @param {Array} stack
+ * @param {Object} props
  * @returns {Array}
  */
-function getReducerProps (createReducer, source, path = [], props = []) {
+function getProps (createReducer, source, stack = [], props = newProps()) {
   for (let key of Object.keys(source)) {
+    const path = stack.concat(key)
     const value = source[key]
-    const fullPath = path.concat(key)
     if (isPlainObject(value)) {
       // NOTE: recursive call
-      getReducerProps(createReducer, value, fullPath, props)
+      getProps(createReducer, value, path, props)
       continue
     }
 
-    props.push({
-      path: fullPath,
-      reducer: createReducer(value)
-    })
+    const reducer = createReducer(value)
+    if (reducer.type === 'ReducerConstant') {
+      set(props.constants, path, reducer.value)
+    } else {
+      props.reducers.push({ path, reducer })
+    }
   }
 
   return props
 }
 
-module.exports.getReducerProps = getReducerProps
+module.exports.getProps = getProps
 
 /**
  * @param {Function} createReducer
  * @param {Object} source
  * @returns {reducer}
  */
 function create (createReducer, source = {}) {
+  const props = getProps(createReducer, source)
```

Comment:
`ReducerObject` now has a `reducers` property and a `source` property.

`source` is a function that returns an object with all the values that were wrapped in `constant`

`reducers` is an array with an entry for every value that's a reducer

```js
const reducer = createReducerObject({
  a: constant(1),
  b: {
    c: constant(2)
  },
  c: () => 'this is a reducer'
})

// constant.source() will return this object:
{
  a: 1,
  b: {
    c: 2
  }
}

// ...and constant.reducers will have information for resolving () => 'this is a reducer'
```

Reply:
this is great! thanks for clarifying!!